aodntools/timeseries_products/gridded_timeseries.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import bisect
+import argparse
+import os.path
+import json
+from datetime import datetime
+
+import xarray as xr
+import pandas as pd
+
+import aggregated_timeseries as TStools
+
+
+TEMPLATE_JSON = 'gridded_timeseries_template.json'
+
+
+def make_depth_bins(nc, increment=10):
+    """
+    generate a list of depth bins from min/max DEPTH.
+    Re-scale the min/max to nearest increment round number and start the bin from there
+    :param nc: hourly dataset
+    :return: list of depth bins
+    """
+
+    dmin = int(nc.DEPTH.min() / increment) * increment
+    dmax = int(nc.DEPTH.max() / increment) * increment + increment
+    dbin = list(range(dmin, dmax, increment))
+
+    return dbin
+
+def adjust_depth_bins(depth_bins, depth_min, depth_max):
+    """
+    Adjust the provided depth bins to match the min and max registered depth
+    :param depth_bins: list of target depths
+    :param depth_min: min valid depth
+    :param depth_max: max valid depth
+    :return: list of adjusted target depths
+    """
+
+    target_depth_min = bisect.bisect_left(depth_bins, depth_min)
```

Comment:
You use bisect only two times and for very simple stuff - finding the closest index of a depth value.
IMO, I don't think that using an external dependency for that is required.

Although the numpy way is a bit verbose, a bisect function is a one liner:

```
def bisect_left(alist,value):
return np.where(np.array(x)<=value)[0][-1]
```

bonus are: faster and one less dependency.

Reply:
I've experimented with bisect and custom made bisect functions as you suggested, but the bisect package resulted way more faster that the custom made bisect functions:

with the `bisect` package
```
In [21]: %timeit get_depth_mask(depth_bins, depths, max_separation)
2.16 µs ± 19.7 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```
with the custom-made functions
```
In [20]: %timeit get_depth_mask2(depth_bins, depths, max_separation)
15.9 µs ± 178 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

```

Reply:
I read the units wrongly then. I just notice that its from the standard library - so disregard the dependency and the comment above